### ISSUE TITLE
ERROR WHILE BUILD SDK

### ISSUE THREAD

**Issue by patadia** [alexa/avs-device-sdk#1136]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:

from a guide for  macOs quickstart in step Build SDK I got following error

p.p1 {margin: 0.0px 0.0px 0.0px 0.0px; font: 12.0px 'Helvetica Neue'}

jays-MacBook-Air:sdk-build jaysanjay$ make SampleApp -j2 >>jay.txt

In file included from /Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/CapabilityAgents/DoNotDisturb/src/DoNotDisturbCapabilityAgent.cpp:16:

In file included from /Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/ThirdParty/rapidjson/rapidjson-1.1.0/include/rapidjson/error/en.h:18:

In file included from /Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/ThirdParty/rapidjson/rapidjson-1.1.0/include/rapidjson/error/../error/error.h:18:

In file included from /Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/ThirdParty/rapidjson/rapidjson-1.1.0/include/rapidjson/error/../internal/../rapidjson.h:144:

In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/string:477:

In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/string_view:176:

In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/__string:56:

In file included from /Library/Developer/CommandLineTools/usr/include/c++/v1/algorithm:643:

/Library/Developer/CommandLineTools/usr/include/c++/v1/memory:2285:5: error: delete called on non-final

      'alexaClientSDK::settings::Setting<bool>' that has virtual functions but non-virtual destructor

      [-Werror,-Wdelete-non-virtual-dtor]

    delete __ptr;

    ^

/Library/Developer/CommandLineTools/usr/include/c++/v1/memory:2598:7: note: in instantiation of member function

      'std::__1::default_delete<alexaClientSDK::settings::Setting<bool> >::operator()' requested here

      __ptr_.second()(__tmp);

      ^

/Library/Developer/CommandLineTools/usr/include/c++/v1/memory:2552:19: note: in instantiation of member function

      'std::__1::unique_ptr<alexaClientSDK::settings::Setting<bool>, std::__1::default_delete<alexaClientSDK::settings::Setting<bool> >

      >::reset' requested here

  ~unique_ptr() { reset(); }

                  ^

/Library/Developer/CommandLineTools/usr/include/c++/v1/memory:4013:21: note: in instantiation of member function

      'std::__1::unique_ptr<alexaClientSDK::settings::Setting<bool>, std::__1::default_delete<alexaClientSDK::settings::Setting<bool> >

      >::~unique_ptr' requested here

    unique_ptr<_Yp> __hold(__p);

                    ^

/Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/Settings/include/Settings/Setting.h:137:9: note: in instantiation of function

      template specialization 'std::__1::shared_ptr<alexaClientSDK::settings::Setting<bool>

      >::shared_ptr<alexaClientSDK::settings::Setting<bool> >' requested here

        std::shared_ptr<Setting<ValueT>>(new Setting(defaultValue, applyValueFn, std::move(settingProtocol)));

        ^

/Users/jaysanjay/sdk-folder/sdk-source/avs-device-sdk/CapabilityAgents/DoNotDisturb/src/DoNotDisturbCapabilityAgent.cpp:143:49: note: 

      in instantiation of member function 'alexaClientSDK::settings::Setting<bool>::create' requested here

    m_dndModeSetting = settings::Setting<bool>::create(false, std::move(protocol));

                                                ^

1 error generated.

make[3]: *** [CapabilityAgents/DoNotDisturb/src/CMakeFiles/DoNotDisturbCA.dir/DoNotDisturbCapabilityAgent.cpp.o] Error 1

make[2]: *** [CapabilityAgents/DoNotDisturb/src/CMakeFiles/DoNotDisturbCA.dir/all] Error 2

make[1]: *** [SampleApp/src/CMakeFiles/SampleApp.dir/rule] Error 2

make: *** [SampleApp] Error 2

jays-MacBook-Air:sdk-build jaysanjay$
 
## Provide the steps to reproduce the issue, if applicable:

from MacOS QUICKSTART GUIDE 
### What version of the AVS Device SDK are you using?
      1.10

### Tell us what hardware you're using:
- [-] Desktop / Laptop
- [ ] Raspberry Pi
- [ ] Other - tell us more:

### Tell us about your OS (Type & version):
- [ ] Linux
- [-] MacOS 
- [ ] Raspbian Stretch
- [ ] Raspbian Jessy
- [ ] Other - tell us more:

**Comment by scotthea-amazon** [comment]
Hello patadia,

Thank you for bringing this to our attention!

From the error messages, it looks like this might be addressed by adding a default virtual destructor to [SettingInterface](https://github.com/alexa/avs-device-sdk/blob/master/Settings/include/Settings/SettingInterface.h#L50).

We did not run across this in our testing on MacOS.  I wonder if your compiler is just configured differently.  Can you attach a copy of CMakeFiles/[your CMake's version]/CMakeCXXCompiler.cmake
from your build folder so that we might better understand this problem?

Best regards,
-SWH

**Comment by frivas** [comment]
Hello @scotthea-amazon, I am also having the same error, Please find enclosed the CMakeCXXCompiler.cmake.

[CMakeCXXCompiler.cmake.zip](https://github.com/alexa/avs-device-sdk/files/2714994/CMakeCXXCompiler.cmake.zip)

Thank you very much in advanced.

**Comment by frivas** [comment]
Hello, 

Just an update. 

I have added the default virtual destructor to the ```SettingInterface.h``` like so:

```c
virtual ~SettingInterface() = default;
```

The build process _almost_ finishes correctly. 

There is an error at the end

```bash
[100%] Building CXX object SampleApp/src/CMakeFiles/SampleApp.dir/main.cpp.o
[100%] Building CXX object SampleApp/src/CMakeFiles/SampleApp.dir/PortAudioMicrophoneWrapper.cpp.o
make[2]: *** No rule to make target `/Users/fran/Downloads/ExplorationProjects/AVS_macOS/sdk-source/avs-device-sdk/third-party/portaudio/lib/.libs/libportaudio.a', needed by `SampleApp/src/SampleApp'.  Stop.
make[1]: *** [SampleApp/src/CMakeFiles/SampleApp.dir/all] Error 2
make: *** [all] Error 2
```

Still investigating what is going on.

Thank you very much again.

**Comment by jbfitbit** [comment]
@visayamv Is there a timeline for that? It's been over two months since the issue was reported.

**Comment by kclchan** [comment]
Thanks for opening this issue.  It has been fixed in our v1.12 release.  Thanks!
